File: 3612-adjacent-increasing-subarrays-detection-i/adjacent-increasing-subarrays-detection-i.java
class Solution {
    public boolean hasIncreasingSubarrays(List<Integer> nums, int k) {
        HashSet<Integer> st = new HashSet<>();
        int cnt = 1;
        if(k == 1) return true;
        for(int i=0; i<nums.size()-1; i++){
            if(nums.get(i) < nums.get(i+1)){
                cnt++;
            }
            else{
                cnt=1;
            }
            if(cnt - k >= 0) st.add(i+1);
        }
        for (int rec : st) {
            if(st.contains(rec - k)){
                return true;
            }
        }
        return false;
    }
}

File: 3612-adjacent-increasing-subarrays-detection-i/adjacent-increasing-subarrays-detection-i.py
class Solution:
    def hasIncreasingSubarrays(self, nums: List[int], k: int) -> bool:
        cnt = 1
        n = len(nums)
        records = []
        if k == 1:
            return True
        for i in range(n-1):
            if nums[i] < nums[i+1]:
                cnt+=1
            else:
                cnt=1
            if cnt - k >= 0:
                records.append(i+1)
        for rec in records:
            if rec - k in records:
                return True
        print(records)
        return False